File: src/main/java/bruteforce/BJ15654.java
package bruteforce;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BJ15654 {
	static int[] arr, ans, visit;
	static int n,m;
	static StringBuilder sb = new StringBuilder();
	
	public static void dfs( int depth ) {
		if( depth == m ) {
			for( int val : ans ) sb.append( val ).append( " " );
			sb.append( "\n" );
			return;
		}
		
		for( int i=0; i<n; i++ ) {
			if( visit[i] != 1 ) {
				ans[depth] = arr[i];
				visit[i] = 1;
				dfs( depth+1 );
				visit[i] = 0;
			}
		}
	}

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer( br.readLine() );
		n = Integer.parseInt( st.nextToken() );
		m = Integer.parseInt( st.nextToken() );
		
		arr = new int[n];
		ans = new int[m];
		visit = new int[n];
		
		st = new StringTokenizer( br.readLine() );
		for( int i=0; i<n; i++ ) arr[i] = Integer.parseInt( st.nextToken() );
		Arrays.sort( arr );
		
		dfs( 0 );
		
		System.out.println( sb );
	}
}

File: Python/bruteforce/BJ15654.py
from sys import stdin

def dfs( depth ):
    global n, m, arr, ans
    if depth == m:
        print( ' '.join( map(str, ans) ) )
        return
    
    for i in range( 0, n ):
        if visit[i]:
            continue
        else:
            ans[depth] = arr[i]
            visit[i] = True
            dfs( depth+1 )
            visit[i] = False
        

input = stdin.readline
n, m = map( int, input().split() )
ans = [0] * m
visit = [False] * n 

arr = list( map(int, input().split()) )
arr.sort()

dfs( 0 )